### PR TITLE
Rename CRDS SnapshotHash to SnapshotHashes

### PR DESCRIPTION
**Pull request by brooksprumo**
In future PRs, I'll be adding new types for handling snapshot hashes. One will be an actual alias (or struct) for `SnapshotHash`. I'll be using that in bootstrap, snapshot packager service, and gossip. So, to avoid confusion, I'm renaming the `SnapshotHash` in CRDS to be `SnapshotHashes`.

Related to #20423

### CODE REVIEW

File: gossip/src/crds_value.rs [hunk 1]
```diff
@@ -84,8 +84,8 @@ pub enum CrdsData {
     ContactInfo(ContactInfo),
     Vote(VoteIndex, Vote),
     LowestSlot(/*DEPRECATED:*/ u8, LowestSlot),
-    SnapshotHashes(SnapshotHash),
-    AccountsHashes(SnapshotHash),
+    SnapshotHashes(SnapshotHashes),
+    AccountsHashes(SnapshotHashes),
```
Comment (brooksprumo) [inline gossip/src/crds_value.rs:88]: In this PR comment (https://github.com/solana-labs/solana/pull/20374#discussion_r721400651), Behzad mentioned that following the naming scheme of (old) `SnapshotHashes` wasn't the right call. So here's the update to make this better.

File: gossip/src/crds_value.rs [hunk 2]
```diff
@@ -160,13 +160,13 @@ impl CrdsData {
 }
 
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, AbiExample)]
-pub struct SnapshotHash {
+pub struct SnapshotHashes {
     pub from: Pubkey,
     pub hashes: Vec<(Slot, Hash)>,
     pub wallclock: u64,
 }
```
Comment (brooksprumo) [inline gossip/src/crds_value.rs:167]: And here's the actual struct. The singular "hash" was a little strange to me, since there's a vector of 'em.

**Review by mvines — APPROVED** [review]
yay, thank you.  The old name has bugged me forever

**Comment by brooksprumo** [comment]
> yay, thank you. The old name has bugged me forever

Happy to do so!

@mvines Also, looks like the frozen abi digest changed (test failed). This is another case where changing the digest hash is OK, yes?

**Comment by mvines** [comment]
> @mvines Also, looks like the frozen abi digest changed (test failed). This is another case where changing the digest hash is OK, yes?

Yep.  The name here doesn't make it into the ABI.  

But careful in general, some types names do make it into the RPC interface.  Those types usually contain a `#[serde(rename_all = "camelCase")]` directive though.

**Review by behzadnouri — APPROVED** [review]

**Pull request merged**